src/main/java/ch/njol/skript/classes/EnumClassInfo.java

```diff
@@ -0,0 +1,68 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.classes;
+
+import ch.njol.skript.expressions.base.EventValueExpression;
+import ch.njol.skript.lang.DefaultExpression;
+import ch.njol.skript.lang.ParseContext;
+import ch.njol.skript.util.EnumUtils;
+import org.eclipse.jdt.annotation.Nullable;
+
+public class EnumClassInfo<T extends Enum<T>> extends ClassInfo<T> {
```

Comment:
Add Javadocs

Reply:
what should it be

Reply:
Something describing what it's used for, what it does and how to use it